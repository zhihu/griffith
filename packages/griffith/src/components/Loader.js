import React, {Component} from 'react'
import {css} from 'aphrodite/no-important'

import styles from './Loader.styles'

class Loader extends Component {
  render() {
    return (
      <div className={css(styles.root)}>
        <svg viewBox="0 0 64 64" className={css(styles.svg)}>
          <circle className={css(styles.circle)} />
        </svg>
      </div>
    )
  }
}

export default Loader
