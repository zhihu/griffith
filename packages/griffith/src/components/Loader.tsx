import React from 'react'
import {css} from 'aphrodite/no-important'

import styles from './Loader.styles'

const Loader = () => {
  return (
    <div className={css(styles.root)}>
      <svg viewBox="0 0 64 64" className={css(styles.svg)}>
        <circle className={css(styles.circle)} />
      </svg>
    </div>
  )
}

export default Loader
