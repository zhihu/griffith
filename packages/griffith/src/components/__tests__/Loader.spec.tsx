import React from 'react'
import Renderer from 'react-test-renderer'
import Loader from '../Loader'

describe('Loader', () => {
  it('get Loader component', () => {
    expect(Renderer.create(<Loader />).toJSON()).toMatchSnapshot()
  })
})
