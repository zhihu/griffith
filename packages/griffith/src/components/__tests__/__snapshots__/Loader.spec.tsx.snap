// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader get Loader component 1`] = `
<div
  className="root_3a2vkm"
>
  <svg
    className="svg_1v9ybk"
    viewBox="0 0 64 64"
  >
    <circle
      className="circle_11txi2m"
    />
  </svg>
</div>
`;
